charms/knative-serving/tests/unit/test_charm.py

```diff
@@ -186,6 +190,35 @@ def test_context_changes(harness):
         assert harness.charm._context == context
 
 
+@pytest.mark.parametrize(
+    "custom_image_config, expected_custom_images",
+    [
+        (
+            yaml.dump({"name1": "image1", "name2": "image2"}),
+            {"name1": "image1", "name2": "image2"},
+        ),
+        (
+            yaml.dump({}),
+            {},
+        ),
+    ],
+)
+def test_custom_images_config_context(custom_image_config, expected_custom_images, harness):
+    """Asserts that the custom_images context is as expected.
+
+    Note: This test is trivial now, where custom_image_config always equals custom_images.  But
+    this will not be the case when we set the DEFAULT_IMAGES dict in the Charm to have our own
+    custom images.
+    """
+    harness.update_config({"custom_images": custom_image_config})
+    harness.begin()
+
+    actual_context = harness.charm._context
+    actual_custom_images = actual_context["custom_images"]
+
+    assert actual_custom_images == expected_custom_images
+
```

Comment:
I'd add test case that asserts we are actually going into `BlockedStatus` when `ValueError` is raised.

Reply:
Added `test_custom_images_config_context_with_incorrect_config` to handle this